### ISSUE TITLE
About environmental variables

### ISSUE THREAD

**Issue by afernandezody** [QMCPACK/qmcpack#2729]
Hello,
I was wondering if there is any environmental variable similar to MKL_ROOT (or HDF5_ROOT) to set up the ARMPL path. I tried ARMPL_ROOT=/opt/arm/armpl_20.3_gcc-9.3 but the CMakeCache.txt still cannot find the libraries.
```
...
//Path to a library
BLAS_acml_mp_LIBRARY:FILEPATH=BLAS_acml_mp_LIBRARY-NOTFOUND

//Path to a library.
BLAS_armpl_lp64_LIBRARY:FILEPATH=BLAS_armpl_lp64_LIBRARY-NOTFOUND

//Path to a library.
BLAS_blas_LIBRARY:FILEPATH=BLAS_blas_LIBRARY-NOTFOUND
...
```
and cmake therefore complains that it cannot find BLAS. Thank you.

**Comment by ye-luo** [comment]
Doesn't the recipe in the manual work for you?
https://qmcpack.readthedocs.io/en/develop/installation.html?highlight=performance%20test#installing-on-systems-with-armv8-based-processors

**Comment by afernandezody** [comment]
Hi @**ye-luo**,
Thanks. I followed the directions on the manual but it's producing the same error with cmake (just for clarification, my compilers are GNU not arm). 
Additionally, I tried the Gitlab package (v3.7.0): the cmake part went through and the compilation almost (got to 85% but began to complain `/opt/arm/armpl_20.3_gcc-9.3/lib/libarmpl_lp64_mp.so: undefined reference to _gfortran_transfer_character_write`).

**Comment by ye-luo** [comment]
This indicate the ARM performance library has extra dependency on gfortran runtime libraries. It assumes gfortran used as the final linker but QMCPACK uses g++. ARM performance library documentation
https://developer.arm.com/documentation/101004/2030/General-Information/Accessing-the-Library
says you need to add 
-lgfortran or -larmflang depending on g++ or armclang++

Try add it to LAPACK_LIBRARIES or/and FFTW_LIBRARIES after the libarmpl

**Comment by afernandezody** [comment]
Hi. You're completely right. I had used the Fortran flag with other apps but dropped it for the QMCPACK build. After some minor tweaking, I was able to compile and the serial code seems to be running. The MPI version was breaking with memory issues (my problem is kind of large) even though there still should be enough memory (I'm assuming that this issue is unrelated but still not 100% sure about it).
